### PR TITLE
eliminate lite-specific copies of sources

### PR DESCRIPTION
**Pull request by sezero**
This patchset eliminates lite-specific format.c and mod_load.c sources.
Closes: https://github.com/libxmp/libxmp/issues/594

@AliceLR, @cmatsuoka: Please review; see https://github.com/libxmp/libxmp/issues/594 for original issue
and several notes.

### CODE REVIEW

File: src/format.h
Comment (ccawley2011) [inline src/format.h:93]:
```suggestion
#elif !defined(LIBXMP_CORE_DISABLE_IT)
#define NUM_FORMATS 4
#define NUM_PW_FORMATS 0
#else
#define NUM_FORMATS 3
#define NUM_PW_FORMATS 0
#endif
```

Also, it might be a good idea to move `NUM_PW_FORMATS` into the `LIBXMP_NO_PROWIZARD` ifdef instead.

Comment (sezero) [reply src/format.h]: Pushed 3ed45bede77bcdcf76f38c16ad023bb66b140227 based on your suggestiontion.

How does it look this time?

File: src/format.c
Comment (AliceLR) [inline src/format.c:39]: IMO just reorder IT and S3M, it doesn't need to be this messy.

File: src/loaders/mod_load.c
Comment (AliceLR) [inline src/loaders/mod_load.c:98]: These should probably be non-indented for consistency with everywhere else? Several other locations too...

**Review by AliceLR — COMMENTED** [review]
Here's a couple of possible minor tidying changes. I don't really have any objections here but Claudio should probably review.

File: src/format.c
Comment (sezero) [reply src/format.c:39]: I didn't want to change the order just in case, but I'll happily do it.

File: src/loaders/mod_load.c
Comment (sezero) [reply src/loaders/mod_load.c:98]: Looked ugly without indented. If you feel strongly about it, I'll de-indent them.

File: src/format.c
Comment (sezero) [reply src/format.c]: And done: 91d1e57ecb80fc2cef5b87881ab082d3248c8e24

**Comment by sezero** [comment]
Rebased.

For Claudio's review: Here are the small differences to what we had before:

- format_loaders[] array moves flt and st loaders down the list to
  reduce ifdefs
- Lite version of mod_test() now tests for `*buf - '0'` for `CHN?` mods
  instead of `*buf >= '0' && *buf <= '9'`
- The lite version now follows full's same style for magic zeroing & read,
  i.e. flips the order of size and nelem params to hio_read(), uses sizeof
  instead of hard-coded 8 in memset.
- The lite version now checks hio_error(f) after magic read just like
  the full version (lack of it in lite was possibly an oversight?)
- The lite version now has the (mod->chn == 0) check from full version.
  Redundant but kept for ifdef simplicity.
- The lite version now has the `mod->chn >= XMP_MAX_CHANNELS` check from
  the full version. Possibly redundant but kept for ifdef simplicity.
- Lite version now has https://github.com/libxmp/libxmp/commit/7cfbc5e21aeda5f0f1a729c3482d8960b91a621f applied.
- The full version adds missing `\n` to a `D_(D_INFO` format string.
- Removes commented `pat_size = ...` statements from the full version.
- Slightly reorders the local vars in mod_load() for simpler ifdefs.
- Removes commented of pat_size = ... from the full version.

**Comment by sezero** [comment]
PING @cmatsuoka ?

File: src/loaders/mod_load.c
Comment (ccawley2011) [inline src/loaders/mod_load.c:499]:
```suggestion
	#ifdef LIBXMP_CORE_PLAYER
```

Comment (sezero) [reply src/loaders/mod_load.c]: Thanks for catching it. Fixed and force-pushed.

**Comment by sezero** [comment]
PING?

**Comment by sezero** [comment]
PING? Pretty please?

File: lite/Makefile.os2
Comment (cmatsuoka) [inline lite/Makefile.os2:4]: If we won't have a separate -lite package, we could just generate the lib with the same file name, I don't think someone would need both versions to be available in the same system. Ideally we could have the lite configuration in the same makefile.

File: lite/Makefile.w32
Comment (cmatsuoka) [inline lite/Makefile.w32:4]: Same here, instead of generating the normal and lite library with two different names it could be some kind of configuration in the same Makefile once all licensing issues are sorted out and the -lite package doesn't need to be published separately.

**Review by cmatsuoka — APPROVED** [review]
Consolidating the sources is good, and we can consider switching between full or lite using a configuration option (or makefile option for certain platforms that don't use a configuration script) when the license switch is done. In this case we could just generate libxmp.dll/libxmp.so in both cases, it would be the same lib but built with different options.

File: src/loaders/mod_load.c
Comment (AliceLR) [inline src/loaders/mod_load.c:664]: `D_()` already injects a `\n`, I think this line is safe to revert.

File: src/loaders/mod_load.c
Comment (AliceLR) [inline src/loaders/mod_load.c:698]: I think it would be better if -lite also used the buffer rather than one `hio_read` per event—this was an optimization we applied a while back to the main MOD loader but not to the -lite MOD loader for some reason. This *might* clean up these a bit, but I can also send a second patch after we merge this if you don't want to.

File: lite/Makefile.os2
Comment (sezero) [reply lite/Makefile.os2:4]: That would be most ideal and it'd reduce maintenance cost.  But it'll have to be a subject of another patchset.

File: lite/Makefile.w32
Comment (sezero) [reply lite/Makefile.w32:4]: That would be most ideal and it'd reduce maintenance cost.  But it'll have to be a subject of another patchset.

File: src/loaders/mod_load.c
Comment (sezero) [reply src/loaders/mod_load.c:698]: Please send a patch for this

File: src/loaders/mod_load.c
Comment (sezero) [reply src/loaders/mod_load.c:664]: Ehh, forgot removing this one, sorry. Can you send a patch packed with the hio_read buffer thing below for this?

**Pull request merged**
